velocity/src/main/java/me/confuser/banmanager/velocity/VelocityServer.java

```diff
@@ -0,0 +1,237 @@
+        break;
+      case "PlayerReportDeletedEvent":
+        event = new PlayerReportDeletedEvent((PlayerReportData) args[0]);
+        break;
+
+      case "NameBanEvent":
+        event = new NameBanEvent((NameBanData) args[0], (boolean) args[1]);
+        break;
+      case "NameBannedEvent":
+        event = new NameBannedEvent((NameBanData) args[0], (boolean) args[1]);
+        break;
+      case "NameUnbanEvent":
+        event = new NameUnbanEvent((NameBanData) args[0], (PlayerData) args[1], (String) args[2]);
+        break;
+
+      case "PlayerWarnEvent":
+        event = new PlayerWarnEvent((PlayerWarnData) args[0], (boolean) args[1]);
+        break;
+      case "PlayerWarnedEvent":
+        event = new PlayerWarnedEvent((PlayerWarnData) args[0], (boolean) args[1]);
+        break;
+
+      case "IpRangeBanEvent":
+        event = new IpRangeBanEvent((IpRangeBanData) args[0], (boolean) args[1]);
+        break;
+      case "IpRangeBannedEvent":
+        event = new IpRangeBannedEvent((IpRangeBanData) args[0], (boolean) args[1]);
+        break;
+      case "IpRangeUnbanEvent":
+        event = new IpRangeUnbanEvent((IpRangeBanData) args[0], (PlayerData) args[1], (String) args[2]);
+        break;
+
+      case "PlayerMuteEvent":
+        event = new PlayerMuteEvent((PlayerMuteData) args[0], (boolean) args[1]);
+        break;
+      case "PlayerMutedEvent":
+        event = new PlayerMutedEvent((PlayerMuteData) args[0], (boolean) args[1]);
+        break;
+      case "PlayerUnmuteEvent":
+        event = new PlayerUnmuteEvent((PlayerMuteData) args[0], (PlayerData) args[1], (String) args[2]);
+        break;
+    }
+
+    if (event == null) {
+      plugin.getLogger().warning("Unable to call missing event " + name);
+
+      return commonEvent;
+    }
+
+    server.getEventManager().fire(event);
+
+    if (event instanceof SilentCancellableEvent) {
+      commonEvent = new CommonEvent(((SilentCancellableEvent) event).isCancelled(), ((SilentCancellableEvent) event).isSilent());
+    } else if (event instanceof SilentEvent) {
+      commonEvent = new CommonEvent(false, ((SilentEvent) event).isSilent());
+    } else if (event instanceof CustomCancellableEvent) {
+      commonEvent = new CommonEvent(((CustomCancellableEvent) event).isCancelled(), true);
+    }
+
+    return commonEvent;
+  }
+
+  public static @NotNull Component formatMessage(String message) {
+    return LegacyComponentSerializer.legacy('&').deserialize(message);
+  }
+
+  public static Component convert(me.confuser.banmanager.common.kyori.text.Component message) {
+    String gson =  me.confuser.banmanager.common.kyori.text.serializer.gson.GsonComponentSerializer.gson().serialize(message);
+    return GsonComponentSerializer.gson().deserialize(gson);
+  }
+
+  @Override
+  public CommonExternalCommand getPluginCommand(String commandName) {
+    // @TODO Seems like Velocity doesn't expose an easy way to retrieve a command by name?
```

Comment:
Is there no way to implement this? This means the blacklist for muted commands will not pick up aliases